diff --git a/apps/openmw/engine.cpp b/apps/openmw/engine.cpp
index 55b8a8f682..a634cec24f 100644
--- a/apps/openmw/engine.cpp
+++ b/apps/openmw/engine.cpp
@@ -7,6 +7,7 @@
 
 #include <osgDB/ReaderWriter>
 #include <osgDB/Registry>
+#include <osgViewer/Viewer>
 #include <osgViewer/ViewerEventHandlers>
 
 #include <SDL.h>
@@ -85,6 +86,9 @@
 
 #include "profile.hpp"
 
+// To share the viewer with Android interfaces
+osg::ref_ptr<osgViewer::Viewer> g_viewer;
+
 namespace
 {
     void checkSDLError(int ret)
@@ -693,6 +697,9 @@ void OMW::Engine::createWindow()
 
     mViewer->getEventQueue()->getCurrentEventState()->setWindowRectangle(
         0, 0, graphicsWindow->getTraits()->width, graphicsWindow->getTraits()->height);
+
+    // To share the viewer with Android interfaces
+    g_viewer = mViewer;
 }
 
 void OMW::Engine::setWindowIcon()
@@ -1074,6 +1081,8 @@ void OMW::Engine::go()
     Settings::Manager::saveUser(mCfgMgr.getUserConfigPath() / "settings.cfg");
     Settings::ShaderManager::get().save();
     mLuaManager->savePermanentStorage(mCfgMgr.getUserConfigPath());
+
+    g_viewer.release();
 }
 
 void OMW::Engine::setCompileAll(bool all)
diff --git a/components/debug/debugging.cpp b/components/debug/debugging.cpp
index ff0404f0d5..054012951d 100644
--- a/components/debug/debugging.cpp
+++ b/components/debug/debugging.cpp
@@ -18,6 +18,13 @@
 
 #include <components/crashcatcher/crashcatcher.hpp>
 #include <components/files/conversion.hpp>
+
+#if defined(__ANDROID__)
+#include <osgViewer/Viewer>
+
+// To share the viewer with Android interfaces; defined in apps/openmw/engine.cpp.
+extern osg::ref_ptr<osgViewer::Viewer> g_viewer;
+#endif
 #include <components/misc/strings/conversion.hpp>
 #include <components/misc/strings/lower.hpp>
 
@@ -482,6 +489,9 @@ namespace Debug
         }
         catch (const std::exception& e)
         {
+#if defined(__ANDROID__)
+            g_viewer.release();
+#endif
 #if (defined(__APPLE__) || defined(__linux) || defined(__unix) || defined(__posix))
             if (!isatty(fileno(stdin)))
 #endif
